subsys/logging/log_backend_rtt.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2018 omSquare s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log_backend.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include <rtt/SEGGER_RTT.h>
+
+
+#if CONFIG_LOG_BACKEND_RTT_MODE_DROP
+
+#define DROP_MESSAGE "\nmessages dropped: 99 \r"
+#define DROP_MESSAGE_LEN (sizeof(DROP_MESSAGE) - 1)
+static const char *drop_msg = DROP_MESSAGE;
+static int drop_cnt = 0;
+static int drop_warn = 0;
+
+#else
+
+#define DROP_MESSAGE_LEN 0
+
+#endif /* CONFIG_LOG_BACKEND_RTT_MODE_DROP */
+
+#if CONFIG_LOG_BACKEND_RTT_BUFFER > 0
+
+#define RTT_LOCK()
```

Comment:
please comment why when BUFFER>0 then lock is not needed.

Reply:
Because rtt backend then use differend non-shared memory region.